### PR TITLE
completes ticket 4 get articles by id

### PR DESCRIPTION
**Pull request by Nicolakirk**

### CODE REVIEW

File: controllers/article_controller.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+const { selectArticles } = require("../models/article_model");
+
+
+exports.getArticles = (req, res, next) =>{
+ 
+    const { article_id } = req.params;
+
+    selectArticles(article_id).then((output) => {
+        res.status(200).send({ output });
```
Comment (popatre) [inline controllers/article_controller.js:9]: Need to use the key `article`

**Review by popatre — CHANGES_REQUESTED** [review]
Hello. 
Lovely error tests - just a few changes needed please

File: __tests__/app.test.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+                .expect(200)
+                .then(({ body }) => {
+                    expect(body.article.title).toBe(
+                       "Living in the shadow of a great man");
+                       expect(body.article.author).toBe ("butter_bridge");
+                        expect(body.article.article_id).toBe( 1);
+                        expect(body.article.body).toBe("I find this existence challenging");
+                        expect(body.article.topic).toBe ("mitch");
+                    expect(body.article.votes).toBe(100);
+                        expect(body.article.comment_count).toBe("11");
```
Comment (NRMSMITH) [inline __tests__/app.test.js:52]: there's no need for comment count yet :)

File: models/article_model.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+            .then((result) => {
+              if (result.rows.length === 0) {
+                return Promise.reject({
+                  status: 404,
+                  msg: "Article can't be found",
+                });
+              } else {
+                return result.rows[0];
+              }
+            });
```
Comment (NRMSMITH) [inline models/article_model.js:21]: There's no need to do a LEFT JOIN here as there is already an author column

File: output.txt [hunk 4]
```diff
@@ -0,0 +1,6 @@
+You are now connected to database "nc_news" as user "nicolakirk".
```
Comment (NRMSMITH) [inline output.txt:1]: these shouldn't be appearing on your commits - please do not use the command 'git add . ' instead you should be only adding the files that are needed

File: playground.sql [hunk 5]
```diff
@@ -0,0 +1,7 @@
+\c nc_news
+CREATE TABLE author_comments(
+    author_id INT REFEREENCES comments(author_id),
+    topic_
+
+)
```
Comment (NRMSMITH) [inline playground.sql:6]: see above comment ^^

**Review by NRMSMITH — CHANGES_REQUESTED** [review]
A couple of requested changes - please submit another pr once you have done them :)

File: __tests__/app.test.js [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+                .expect(200)
+                .then(({ body }) => {
+                    expect(body.article.title).toBe(
+                       "Living in the shadow of a great man");
+                       expect(body.article.author).toBe ("butter_bridge");
+                        expect(body.article.article_id).toBe( 1);
+                        expect(body.article.body).toBe("I find this existence challenging");
+                        expect(body.article.topic).toBe ("mitch");
+                    expect(body.article.votes).toBe(100);
+
```
Comment (popatre) [inline __tests__/app.test.js:52]: Just looking at the ticket - it asks for:
author
title
article_id
body
topic
created_at
votes
article_img_url

We're missing :

- Created_at - can just expect a string for this.
- article_img_url

File: models/article_model.js [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+
+exports.selectArticles = (id) => {
+  
+           return db.query (  `SELECT articles.*, COUNT(comments.article_id) AS comment_count
+           FROM articles 
+          JOIN COMMENTS
+           ON articles.article_id = comments.article_id 
+           WHERE articles.article_id = $1
+           GROUP BY articles.article_id
+           ;`, [id])
```
Comment (popatre) [inline models/article_model.js:11]: So this query can just be `SELECT * from articles where article_id = $1`

There is no need for a JOIN of any kind. You have all the data needed inside the article table - dont need the comment count for this ticket.

**Review by popatre — CHANGES_REQUESTED** [review]

File: __tests__/app.test.js [hunk 8]
```diff
@@ -37,18 +38,24 @@ const app = require("../app");
...
+                    const { article } = body;
+                    expect(article).toBeInstanceOf(Object);
+                    expect(article).toMatchObject({
+                       title: "Living in the shadow of a great man",
+                       author : "butter_bridge",
+                      article_id : 1,
+                        body:"I find this existence challenging",
+                      topic: "mitch",
+                    votes:100,
+                    created_at:expect.any(String),
```
Comment (popatre) [inline __tests__/app.test.js:55]: Much better 👍

File: models/article_model.js [hunk 9]
```diff
@@ -2,13 +2,9 @@ const db = require("../db/connection");
...
-           return db.query (  `SELECT articles.*, COUNT(comments.article_id) AS comment_count
-           FROM articles 
-          JOIN COMMENTS
-           ON articles.article_id = comments.article_id 
-           WHERE articles.article_id = $1
-           GROUP BY articles.article_id
-           ;`, [id])
+           return db.query (  `
+           SELECT * from articles 
+           where article_id = $1`, [id])
```
Comment (popatre) [inline models/article_model.js:7]: Excellent!

**Review by popatre — APPROVED** [review]
Looks all good to go, well done

**Pull request merged**